Demo/app/src/main/java/com/juanocampo/swipeable/swipeablelist/swpeable/SwipeableWrapperHolder.java

```diff
@@ -0,0 +1,77 @@
+package com.juanocampo.swipeable.swipeablelist.swpeable;
+
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.juanocampo.swipeable.swipeablelist.R;
+import com.juanocampo.swipeable.swipeablelist.backlistener.OnBackListener;
+import com.juanocampo.swipeable.swipeablelist.swpeable.fragment.SwipeableFragment;
+
+/**
+ * Created by juan.ocampo on 09/08/2016.
+ */
+
+public abstract class SwipeableWrapperHolder extends Fragment implements SwipeableFragment.SwipeFragmentAction, OnBackListener {
+
+    private static final int NO_POSITION_CACHED = -1;
+
+    private int lastSelectedPosition = NO_POSITION_CACHED;
+    private SwipeableFragment fragmentSwipeable;
+    private Fragment fragmentNext;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.swipeable_viewswitcher, container, false);
+    }
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        fragmentSwipeable = getSwipeableMainFragment();
+
+        fragmentSwipeable.onAttachSwipeListener(this);
+
+        getActivity().getSupportFragmentManager().beginTransaction().replace(R.id.swipeable_container, fragmentSwipeable)
```

Comment:
This is just for demo proposes, in the real implementation i will use the Navigator, however i will change this :).